dotCMS/src/integration-test/java/com/dotcms/vanityurl/business/VanityUrlAPITest.java

```diff
@@ -60,271 +61,197 @@ public static void prepare() throws Exception {
+     * Creates a couple of vanity urls, publish only the first one.
+     * Checks that the first one is in cache and the second one is not.
      *
      * @see VanityUrlAPI
      */
     @Test
-    public void getVanityUrlFromContentlet() {
-        Contentlet contentlet1 = null;
-        try {
-            long i = System.currentTimeMillis();
-            String title = "VanityURL" + i;
-            String site = defaultHost.getIdentifier();
-            String uri = "/test3_" + i;
-            String forwardTo = "/about-us/" + CMSFilter.CMS_INDEX_PAGE;
-            int action = 301;
-            int order = 1;
-
-            contentlet1 = filtersUtil.createVanityUrl(title, site, uri,
-                    forwardTo, action, order, defaultLanguageId);
-
-            filtersUtil.publishVanityUrl(contentlet1);
-
-            VanityUrl vanity1 = vanityUrlAPI
-                    .getVanityUrlFromContentlet(contentlet1);
-
-            Assert.assertNotNull(vanity1);
-            Assert.assertNotEquals(VanityUrlAPI.CACHE_404_VANITY_URL,vanity1.getInode());
-            Assert.assertEquals(title, vanity1.getTitle());
-            Assert.assertEquals(site, vanity1.getSite());
-            Assert.assertEquals(uri, vanity1.getURI());
-            Assert.assertEquals(forwardTo, vanity1.getForwardTo());
-            Assert.assertEquals(action, vanity1.getAction());
-
-        } catch (DotDataException | DotSecurityException e) {
-            e.printStackTrace();
-            fail();
-        } finally {
-            try {
-                if (contentlet1 != null) {
-                    contentletAPI.destroy(contentlet1, user, false );
-                }
-            } catch (Exception e) {
-                Logger.error(this.getClass(), "Error cleaning up Vanity Url Links");
-            }
-        }
+    public void vanities_resolve_once_they_are_published() throws Exception {
+
+
+ 
+        long i = System.currentTimeMillis();
+        String title = "VanityURL" + i;
+        String site = defaultHost.getIdentifier();
+        String uri = "/" + UUIDGenerator.shorty() + i;
+        String forwardTo = "/products/";
+        int action = 200;
+        int order = 1;
+
+        Contentlet contentlet1 = filtersUtil.createVanityUrl(title, site, uri,
+                forwardTo, action, order, defaultLanguage.getId());
+
+        
+        Optional<CachedVanityUrl> vanityResolved = vanityUrlAPI.resolveVanityUrl(uri, defaultHost, defaultLanguage);
+        
+        // there is no vanity as it is not published
+        assert(!vanityResolved.isPresent());
+        
+        
+        // publish the vanity
+        filtersUtil.publishVanityUrl(contentlet1);
+
+        // resolve again
+        vanityResolved = vanityUrlAPI.resolveVanityUrl(uri, defaultHost, defaultLanguage);
+        
+        // we hava a vanity
+        assert(vanityResolved.isPresent());
+
+        CachedVanityUrl vanity = vanityResolved.get();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'vanity' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)